src/pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,55 @@
+import NextAuth, { NextAuthOptions } from "next-auth";
+import CredentialsProvider from "next-auth/providers/credentials";
+
+import { authorize } from "@server/services/userService";
+
+export const authOptions: NextAuthOptions = {
+  session: {
+    strategy: "jwt",
+  },
+  providers: [
+    CredentialsProvider({
+      name: "Email",
+      credentials: {
+        email: {
+          label: "Email",
+          type: "email",
+          placeholder: "example@sayyara.com",
+        },
+        password: { label: "Password", type: "password" },
+      },
+      async authorize(credentials, _req) {
+        if (!credentials) return null;
+        return await authorize(credentials.email, credentials.password).catch(
+          (_reason) => null
+        );
+      },
+    }),
+  ],
+  callbacks: {
+    jwt({ token, user }) {
```

Comment:
what does the full token and user object look like? I'm wondering if we can use spread `...` notation here

Reply:
The token and user are defined in `src/types/next-auth.d.ts`. They are defined there because I need more field than the default. I don't see what you plan to do with the spread notation?

Reply:
Token and user both use the JWT interface?
If so, can't we do something like token = {...token, user}?

Reply:
I meant if the user object only contains id, first name, and last name, we can use spread notation to override the id, firstname, and last name fields of object from user